src/tasks/task2.js

```diff
@@ -0,0 +1,34 @@
+/* TODO: #2. Функция принимает массив объектов. Исходные данные представлены ниже. Возвращает массив пользователей,
+отсортированных по полю username в алфавитном порядке и возраст которых начинается от 21 года. Количество пользователей
+может быть не ограничено. */
+
+const initialData = [
+    {
+        username: 'Maria',
+        age: 25,
+    },
+    {
+        username: 'Alex',
+        age: 21,
+    },
+    {
+        username: 'Oleg',
+        age: 14,
+    },
+    {
+        username: 'Dmitriy',
+        age: 35,
+    },
+    {
+        username: 'Oksana',
+        age: 72,
+    },
+];
+
+const sortUsersByAge = (usersArray) => {
+    if (!Array.isArray(usersArray)) return;
+    const newArray = usersArray.filter(user => user.age >= 21);
+    return newArray.sort((currentUser, nextUser) => currentUser.username > nextUser.username ? 1 : -1);
```

Comment:
everything is cool, as an improovement I can suggest to remove local variavle `newArray` and just write 
`return usersArray.filter(user => user.age >= 21).sort((currentUser, nextUser) => currentUser.username > nextUser.username ? 1 : -1)`